src/audio/mixer.c

```diff
@@ -245,13 +245,18 @@ static int mixer_trigger(struct comp_dev *dev, int cmd)
 
 	/* don't stop mixer if at least one source is active */
 	if (mixer_source_status_count(dev, COMP_STATE_ACTIVE) &&
-	    (cmd == COMP_TRIGGER_PAUSE || cmd == COMP_TRIGGER_STOP)) {
+	    cmd == COMP_TRIGGER_STOP) {
 		dev->state = COMP_STATE_ACTIVE;
 		ret = PPL_STATUS_PATH_STOP;
 	/* don't stop mixer if at least one source is paused */
 	} else if (mixer_source_status_count(dev, COMP_STATE_PAUSED) &&
 		   cmd == COMP_TRIGGER_STOP) {
-		dev->state = COMP_STATE_PAUSED;
+		dev->state = COMP_STATE_ACTIVE;
+		ret = PPL_STATUS_PATH_STOP;
+	}
+	/* don't stop mixer on pause */
+	if (cmd == COMP_TRIGGER_PAUSE) {
+		dev->state = COMP_STATE_ACTIVE;
```

Comment:
you may want to reflow the logic as

````c
if ((cmd == COMP_TRIGGER_STOP) && 
            (mixer_source_status_count(dev, COMP_STATE_ACTIVE) ||
            mixer_source_status_count(dev, COMP_STATE_PAUSED)) ||
    (cmd == COMP_TRIGGER_PAUSE)) {
                dev->state = COMP_STATE_ACTIVE;
		ret = PPL_STATUS_PATH_STOP;
}
````

or use a switch case with a fallback.




￼

Reply:
@abonislawski ping?

Reply:
@plbossart ok updated, sorry for the delay